test/handlers/FlexVotingClientHandler.sol

```diff
@@ -0,0 +1,235 @@
+  uint256 public ghost_doubleVoteActors;
+
+  constructor(
+    GovToken _token,
+    FractionalGovernor _governor,
+    MockFlexVotingClient _client,
+    ProposalReceiverMock _receiver
+  ) {
+    token = _token;
+    flexClient = _client;
+    governor = _governor;
+    receiver = _receiver;
+  }
+
+  modifier countCall(bytes32 key) {
+    calls[key].count++;
+    _;
+  }
+
+  modifier createActor() {
+    vm.assume(_validActorAddress(msg.sender));
+    currentActor = msg.sender;
+    _actors.add(msg.sender);
+    _;
+  }
+
+  modifier maybeCreateVoter() {
+    if (_proposals.length() == 0) _voters.add(currentActor);
+    _;
+  }
+
+  modifier useActor(uint256 actorIndexSeed) {
+    currentActor = _randAdress(_actors, actorIndexSeed);
+    _;
+  }
+
+  modifier useVoter(uint256 _voterSeed) {
+    currentActor = _randAdress(_voters, _voterSeed);
+    _;
+  }
+
+  function _randAdress(EnumerableSet.AddressSet storage _addressSet, uint256 _seed) internal returns (address) {
+    uint256 len = _addressSet.length();
+    return len > 0 ?  _addressSet.at(_seed % len) : address(0);
+  }
+
+  function lastProposal() external returns (uint256) {
+    return _proposals.at(_proposals.length() - 1);
+  }
+
+  function _randProposal(uint256 _seed) internal returns (uint256) {
+    uint256 len = _proposals.length();
+    return len > 0 ? _proposals.at(_seed % len) : 0;
+  }
+
+  function _validActorAddress(address _user) internal returns (bool) {
+    return _user != address(0) &&
+      _user != address(flexClient) &&
+      _user != address(governor) &&
+      _user != address(receiver) &&
+      _user != address(token);
+  }
+
+  function _remainingTokens() internal returns (uint128) {
+    return type(uint128).max - ghost_mintedTokens;
+  }
+
+  function proposalLength() public returns(uint256) {
+    return _proposals.length();
+  }
+
+  // TODO This always creates a new actor. Should it?
+  function deposit(
+    uint208 _amount
+  ) createActor maybeCreateVoter countCall("deposit") external {
+    vm.assume(_remainingTokens() > 0);
+    _amount = uint208(bound(_amount, 0, _remainingTokens()));
+
+    // Some actors won't have the tokens they need. This is deliberate.
+    if (_amount <= _remainingTokens()) {
+      token.exposed_mint(currentActor, _amount);
+      ghost_mintedTokens += uint128(_amount);
+    }
+
+    vm.startPrank(currentActor);
+    // TODO we're pre-approving every depositi, should we?
+    token.approve(address(flexClient), uint256(_amount));
+    flexClient.deposit(_amount);
+    vm.stopPrank();
+
+    ghost_depositSum += _amount;
+    ghost_accountDeposits[currentActor] += uint128(_amount);
+  }
+
+  // TODO we restrict withdrawals to addresses that have balances, should we?
+  function withdraw(
+    uint256 _userSeed,
+    uint208 _amount
+  )
+    useActor(_userSeed)
+    countCall("withdraw")
+    external
+  {
+    // TODO we limit withdrawals to the total amount deposited, should we?
```

Comment:
Sounding like a broken record here, but if we know a call will definitely revert every time, and we have that case sufficiently unit tested, I think it's ok to bound it out. Fine line, though, and this is a weakly held opinion. I have accidentally bounded out valid failure cases before.